src/satosa/micro_services/processors/scope_remover_processor.py

```diff
@@ -0,0 +1,19 @@
+from ..attribute_processor import AttributeProcessorError, AttributeProcessorWarning
+from .base_processor import BaseProcessor
+
+class ScopeRemoverProcessor(BaseProcessor):
+    """
+    Removes the scope from all values of a given attribute 
+    
+    """
+    def process(self, internal_data, attribute, **kwargs):
+        attributes = internal_data.attributes
+        new_values = []
+        values = attributes.get(attribute, [None])
+        if values:
+            for value in attributes.get(attribute, [None]):
```

Comment:
You already have `values` set, no need to `get()` from `attributes` again ;) 

---

btw, if you want to make this "prettier", you can invert the conditional and remove the else clause and indentation

```python
if not correct:
  raise SomeErrror # processing will stop here

# "implied" else can go here now
do_stuff(...)
do_other_stuff(...)
```

This also moves the error handling closer to the cause, which makes things a bit more readable.